### PR TITLE
Add "chlorophyll" to hofx_3d ctest

### LINKED ISSUE #350: Add simple generic Chl UFO to the 3D h(x) test
## Description
Simulate Chl from soca.

## Solution
- Add a sample viirs chl retrieval obs file
- Add the obsspace to the 3D h(x) test case

## Dependencies
#349 will need to be implemented first.

### PR DESCRIPTION
**Pull request by liuxiao37k**
## Description

This PR adds "chlorophyll" (ocean color) observation operator to hofx_3d ctest in SOCA.
A few generic QC filters are applied in the ctest; also a QC filter specific to L2 ocean color observations using the `any_bit_set` keyword in the `where` statement will be applied but is commented out for now waiting for the merge of https://github.com/JCSDA/ufo/pull/1200. 

### Issue(s) addressed
- fixes #350 

## Testing
The updated hofx_3d has been tested on Hera.

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
Thanks @liuxiao37k , looks good but could the filter be commented out?

**Comment by liuxiao37k** [comment]
> Thanks @liuxiao37k , looks good but could the filter be commented out?

Sorry I was referring to the lines below. Once JCSDA/ufo#1200 is merged we could uncomment these lines but we could also just delete them for the purpose of this test.

       #where:
       #- variable:
       #    name: mass_concentration_of_chlorophyll_in_sea_water@PreQC
       #  any_bit_set_of: 0,1,3,4,5,8,9,10,12,14,15,16,19,21,22,25
       #action:
       #  name: inflate error
       #  inflation factor: 2.0

File: test/testinput/hofx_3d.yml [hunk 1]
```diff
@@ -72,3 +72,36 @@ observations:
...
+          name: mass_concentration_of_chlorophyll_in_sea_water@ObsValue
+        minvalue: 0.001
+        maxvalue: 10.0
+      - variable:
+          name: latitude@MetaData
+        minvalue: -60.0
+        maxvalue:  60.0
+      action:
+        name: reject
+      #where:
```
Comment (guillaumevernieres) [inline test/testinput/hofx_3d.yml]: Add a comment here:
`TODO (Xioa) un-comment below filter when JCSDA/ufo#1200 is merged`
Or something of the same effect.

**Comment by guillaumevernieres** [comment]
> > Thanks @liuxiao37k , looks good but could the filter be commented out?
> 
> Sorry I was referring to the lines below. Once [JCSDA/ufo#1200](https://github.com/JCSDA/ufo/pull/1200) is merged we could uncomment these lines but we could also just delete them for the purpose of this test.
> 
> ```
>    #where:
>    #- variable:
>    #    name: mass_concentration_of_chlorophyll_in_sea_water@PreQC
>    #  any_bit_set_of: 0,1,3,4,5,8,9,10,12,14,15,16,19,21,22,25
>    #action:
>    #  name: inflate error
>    #  inflation factor: 2.0
> ```

Oops, I see that my question is answered in your pr description! Sorry ... Looks good, maybe add a todo comment on top of the filter so we remember why it was commented out (you will remember but I won't).

**Comment by liuxiao37k** [comment]
> > > Thanks @liuxiao37k , looks good but could the filter be commented out?
> > 
> > 
> > Sorry I was referring to the lines below. Once [JCSDA/ufo#1200](https://github.com/JCSDA/ufo/pull/1200) is merged we could uncomment these lines but we could also just delete them for the purpose of this test.
> > ```
> >    #where:
> >    #- variable:
> >    #    name: mass_concentration_of_chlorophyll_in_sea_water@PreQC
> >    #  any_bit_set_of: 0,1,3,4,5,8,9,10,12,14,15,16,19,21,22,25
> >    #action:
> >    #  name: inflate error
> >    #  inflation factor: 2.0
> > ```
> 
> Oops, I see that my question is answered in your pr description! Sorry ... Looks good, maybe add a todo comment on top of the filter so we remember why it was commented out (you will remember but I won't).

@guillaumevernieres The QC PR has been just merged (yay!) so I uncommented these lines in the hofx_3d YAML file.

**Comment by liuxiao37k** [comment]
@jkbk2004 just to make sure that you receive notifications.

**Review by guillaumevernieres — APPROVED** [review]
Looks good, thanks for doing that work @liuxiao37k

**Pull request merged**